src/report.py

```diff
@@ -0,0 +1,98 @@
+import csv
+from typing import List, Dict, Tuple
+
+
+def read_portfolio(filename: str) -> List[Dict]:
+    """
+    Reads a portfolio file and places each entry into a list.
+
+    :param filename: Name of the portfolio file
+    :type filename: str
+    :raises ValueError: Raises ValueError if line in the file is empty
+    :return: Returns list of portfolio entries
+    :rtype: List[Dict]
+    """
+    portfolio = []
+    with open(filename) as f:
+        rows = csv.reader(f)
+        headers = next(rows)
+        for row in rows:
```

Comment:
Here you are iterating over a list of tuple with 3 elements, instead of using `for row in rows` and then use `row[0]`, ... you can do:
```python
for name, share, price in rows:
    portfolio.append(dict(zip(headers, (name, int(shares), float(price))))
```

With this technique, it is easier for the user to understand what is row[0] row[1] and row[2] since we are giving a proper name